tests/datavolume_test.go

```diff
@@ -134,6 +136,254 @@ var _ = Describe("DataVolume Integration", func() {
+				if err != nil {
+					return err
+				}
+				return nil
+			}, 160*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		waitForDeletionVM := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err := virtClient.VirtualMachine(namespace).Get(name, &metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+		}
+
+		waitForDeletionVMI := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err := virtClient.VirtualMachineInstance(namespace).Get(name, &metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+		}
+
+		waitForDeletionDataVolume := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err := virtClient.CdiClient().CdiV1alpha1().DataVolumes(namespace).Get(name, metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+		}
+
+		waitForDeletionPVC := func(name, namespace string) {
+			Eventually(func() bool {
+				_, err = virtClient.CoreV1().PersistentVolumeClaims(namespace).Get(name, metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, 100*time.Second, 1*time.Second).Should(BeTrue())
+
+		}
+
+		AfterEach(func() {
+			if vmJson != "" {
+				err = os.Remove(vmJson)
+				Expect(err).ToNot(HaveOccurred())
+				vmJson = ""
+			}
+
+			deleteIfExistsVM(vm.Name, vm.Namespace)
+			deleteIfExistsVMI(vm.Name, vm.Namespace)
+			deleteIfExistsDataVolume(dataVolumeName, vm.Namespace)
+		})
+
+		It("deleting VM with cascade=true should automatically delete DataVolumes and VMI owned by VM.", func() {
+			By("Creating VM with DataVolumeTemplate entry with k8s client binary")
+			_, _, err = tests.RunCommand(k8sClient, "create", "-f", vmJson)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verifying DataVolume succeeded and is created with VM owner reference")
+			dataVolumeIsSuccessAndOwned(dataVolumeName, vm.Namespace)
+
+			By("Verifying PVC is created")
+			pvcExists(pvcName, vm.Namespace)
+
+			By("Verifying VMI is created with VM owner reference")
+			vmiIsRunningAndOwned(vm.Name, vm.Namespace)
+
+			By("Deleting VM with cascade=true")
+			_, _, err = tests.RunCommand("kubectl", "delete", "vm", vm.Name, "--cascade=true")
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for the VM to be deleted")
+			waitForDeletionVM(vm.Name, vm.Namespace)
+
+			By("Waiting for the VMI to be deleted")
+			waitForDeletionVMI(vm.Name, vm.Namespace)
+
+			By("Waiting for the DataVolume to be deleted")
+			waitForDeletionDataVolume(dataVolumeName, vm.Namespace)
+
+			By("Waiting for the PVC to be deleted")
+			waitForDeletionPVC(pvcName, vm.Namespace)
+		})
+
+		It("deleting VM with cascade=false should orphan DataVolumes and VMI owned by VM.", func() {
```

Comment:
test_id:

It("[test_id:838]deleting VM with cascade=false should orphan DataVolumes and VMI owned by VM.", func() {

Reply:
fixed